server/src/main/java/com/google/coffeehouse/servlets/UpdateClubServlet.java

```diff
@@ -0,0 +1,227 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.gson.GsonFactory;
+import com.google.coffeehouse.common.Book;
+import com.google.coffeehouse.common.Club;
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to update a {@link Club} from Http POST Request Body (in JSON format),
+ * save it in database, and return it in JSON format.
+ */
+@WebServlet("/api/update-club")
+public class UpdateClubServlet extends HttpServlet {
+  /** The fields of the Club object that can be updated. */
+  public static final List<String> updateableClubFields = Arrays.asList(
+      Club.DESCRIPTION_FIELD_NAME,
+      Club.CONTENT_WARNINGS_FIELD_NAME);
+  /** The fields of the Book object that can be updated. */
+  public static final List<String> updateableBookFields = Arrays.asList(
+      Book.AUTHOR_FIELD_NAME,
+      Book.ISBN_FIELD_NAME,
+      Book.TITLE_FIELD_NAME);
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request cannot be parsed for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** Message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE =
+      "Body unable to be parsed in UpdateClubServlet: ";
+  /** Message to be logged when the body of the POST request does not have required fields. */
+  public static final String LOG_INPUT_ERROR_MESSAGE =
+      "Error with JSON input in UpdateClubServlet: ";
+  /**
+   * Message to be logged when an invalid ID token is passed in or a valid ID token that is
+   * associated with a user who does not have the permissions to update the Club is paeed in.
```

Comment:
Nit: passed in

Reply:
Ah, my apologies I really should have caught this before pushing. It has been fixed.